openedx/features/announcements/tests/test_announcements.py

```diff
@@ -0,0 +1,69 @@
+"""
+Unit tests for the announcements feature.
+"""
+
+from django.conf import settings
+from django.test import TestCase, override_settings
+from django.test.client import Client
+
+from student.tests.factories import UserFactory, AdminFactory
+
+from openedx.features.announcements.models import Announcement
+
+TEST_ANNOUNCEMENTS = [
+    ("Active Announcement", True),
+    ("Inactive Announcement", False),
+    ("<strong>Formatted Announcement</strong>", True),
+]
+
+FEATURES = settings.FEATURES.copy()
+FEATURES["ENABLE_ANNOUNCEMENTS"] = True
+@override_settings(FEATURES=FEATURES)
+class TestGlobalAnnouncements(TestCase):
+    """
+    Test Announcements
+    """
+
+    @classmethod
+    def setUpTestData(cls):
+        Announcement.objects.bulk_create([
+            Announcement(content=content, active=active)
+            for content, active in TEST_ANNOUNCEMENTS
+        ])
+
+    def setUp(self):
+        self.client = Client()
+        self.admin = AdminFactory.create(
+            email='staff@edx.org',
+            username='admin',
+            password='pass'
+        )
+        self.client.login(username=self.admin.username, password='pass')
+
+    @override_settings(FEATURES=settings.FEATURES)
+    def test_feature_flag(self):
+        """Ensures that the default settings effectively disables the feature"""
+        response = self.client.get('/dashboard')
+        self.assertNotIn('AnnouncementsView', response.content)
+        self.assertNotIn('<div id="announcements"', response.content)
```

Comment:
Added these tests apart from the tests for Edit and Delete, still working on those.
I hadn't implemented them to begin with since I thought they might change if we move the editor around in the studio anyhow.

Reply:
Scratch that, you just helped me fix the tests for these. I pushed them as well.

Reply:
+1